paddle/fluid/imperative/amp_auto_cast.cc

```diff
@@ -184,6 +184,17 @@ NameVarBaseMap AutoCastInputs(const std::string& op_type,
     return new_ins;
   } else {
     auto dst_type = GetPromoteType(ins);
+
+    if (op_type == "moving_average_abs_max_scale") {
+      for (const auto& pair : ins) {
+        if (pair.first == "X" &&
+            pair.second.front()->DataType() ==
+                framework::proto::VarType::FP16) {
+          dst_type = framework::proto::VarType::FP16;
+        }
+      }
+    }
```

Comment:
can it be handled like "batch_norm", as the code below?

Reply:
moving_average_abs_max_scale only considers the dtype of input(X) to determine the dest_type. However, "batch_norm" just cast the input(X) to FP16 or FP32. These are two different cases.